x-pack/legacy/plugins/siem/public/components/ml/conditional_links/rison_helpers.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RisonValue, RisonObject, decode } from 'rison-node';
+import { isObject, isString } from 'lodash/fp';
+
+export const decodeRison = (value: string): RisonValue => {
+  try {
+    return decode(value);
+  } catch (error) {
+    return null;
+  }
+};
+
+export const isRisonObject = (value: RisonValue): value is RisonObject => {
```

Comment:
Wasn't aware of these [User-Defined Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) -- nice to know, thanks! :)